src/algebra/ring/basic.lean

```diff
@@ -646,6 +661,18 @@ instance ring.to_semiring : semiring α :=
     by rw [← mul_add, add_zero, add_zero],
   ..‹ring α› }
 
+/-- Pullback a `non_unital_non_assoc_ring` instance along an injective function.
+See note [reducible non-instances]. -/
+@[reducible]
+protected def function.injective.non_unital_non_assoc_ring
```

Comment:
This assumes `[ring α]` via the `variables` above, but only needs `[non_unital_non_assoc_ring α]`. Can you move this outside the `ring` section?

Reply:
Thanks, sorry, I should have spotted that. Fixed now, which also fixes the problem I was having above with `equiv`.

Do I need to do anything further for `equiv`?